src/Core/Content/Media/Thumbnail/ThumbnailSizeCalculator.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopware\Core\Content\Media\Thumbnail;
+
+use Shopware\Core\Content\Media\Aggregate\MediaThumbnailSize\MediaThumbnailSizeEntity;
+
+class ThumbnailSizeCalculator
+{
+    /**
+     * @param array{width: int, height: int} $imageSize
```

Comment:
we should specify this as a phpstan type and then reuse it here and for the other method as well
https://phpstan.org/writing-php-code/phpdoc-types#local-type-aliases